### PR TITLE
ARROW-8397: [C++] Fail to compile aggregate_test.cc on Ubuntu 16.04

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit test-ubuntu-16.04-cpp

**Review by bkietz — APPROVED** [review]

File: cpp/src/arrow/compute/kernels/aggregate_test.cc [hunk 1]
```diff
@@ -398,9 +398,9 @@ class TestNumericMinMaxKernel : public ComputeFixture, public TestBase {
     ASSERT_OK(MinMax(&this->ctx_, options, array, &out));
 
     ASSERT_TRUE(out.is_collection());
-    for (const auto& out : out.collection()) {
```
Comment (wesm) [inline cpp/src/arrow/compute/kernels/aggregate_test.cc:401]: :eyes:

**Review by wesm — APPROVED** [review]
+1. It's surprising that this compiles successfully in our other CI entries

**Comment by wesm** [comment]
Ah the crossbow build still needs to run

**Pull request closed without merging**